python/ray/worker.py

```diff
@@ -418,6 +418,17 @@ def put_object(self, object_id, value):
             logger.info(
                 "The object with ID {} already exists in the object store."
                 .format(object_id))
+        except (TypeError, Exception):
```

Comment:
This logic is all in `store_and_register`.

Reply:
No, this PR is handling the case where not all of the members can be serialized with cloudpickle but the overall object can. So we need this extra fallback here.

See the error in https://github.com/ray-project/ray/issues/3373

Reply:
No. It's different. After registering the object with `use_dict=True`, the serializer will skip `__reduce__` method of the original object, and if the object has a strange cython function that cannot be serialized by cloudpickle, the whole process will fail.

Reply:
Can you double check that when you register a custom serializer the second time, it overrides the first one? May be worth adding a test for that.

Reply:
Also, can you document the reason for putting this code here and not in `store_and_register`?

Reply:
Register a custom serializer the second time always updates the serialization context (in pyarrow's source code). The created test has proven that (or it will fail because it still uses use_dict=True).